locale/en/docs/guides/security/index.md

```diff
@@ -0,0 +1,440 @@
+  * Tools such as [`Socket`][] can be used to analyze packages with static analysis
+  to find risky behaviors such as network or filesystem access.
+* Use [`npm ci`][] instead of `npm install`.
+This enforces the lockfile so that inconsistencies between it and the
+_package.json_ file cause an error (instead of silently ignoring the lockfile
+in favor of _package.json_).
+* Carefully check the _package.json_ file for errors/typos in the names of the
+dependencies.
+
+### Memory Access Violation (CWE-284)
+
+Memory-based or heap-based attacks depends on a combination of memory management
+errors and an exploitable memory allocator.
+Like all runtimes, Node.js is vulnerable to these attacks if your projects runs
+on a shared machine.
+Using a secure heap is useful for preventing sensitive information from leaking
+due to pointer overruns and underruns.
+
+Unfortunately, secure heap is not available on Windows.
+More information can be found on Node.js [secure-heap documentation][].
+
+**Mitigations**
+
+* Use `--secure-heap=n` depending on your application where _n_ is the allocated
+maximum byte size.
+* Do not run your production app in a shared machine.
+
+### Monkey Patching (CWE-349)
+
+Monkey patching refers to the modification of properties in runtime aiming to
+change the existing behavior. Example:
+
+```js
+// eslint-disable-next-line no-extend-native
+Array.prototype.push = function (item) {
+  // overriding the global [].push
+};
+```
+
+This is **not** considered a vulnerability, it falls under the following
+statement: _“Node.js trusts the code being run”_.
+
+**Mitigations**
+
+The `--frozen-intrinsics` flag enables experimental[¹][experimental-features]
+frozen intrinsics, which means all the built-in JavaScript objects and functions
+are recursively frozen.
+Therefore, the following snippet **will not** override the default behavior of
+`Array.prototype.push`
+
+```js
+// eslint-disable-next-line no-extend-native
+Array.prototype.push = function (item) {
+  // overriding the global [].push
+};
+
+// Uncaught:
+// TypeError <Object <Object <[Object: null prototype] {}>>>:
+// Cannot assign to read only property 'push' of object ''
+```
+
+However, it’s important to mention you can still define new globals and replace
+existing globals using `globalThis`
+```console
+> globalThis.foo = 3; foo; // you can still define new globals
+3
+> globalThis.Array = 4; Array; // However, you can also replace existing globals
+4
+```
+
+Therefore, `Object.freeze(globalThis)` can be used to guarantee no globals will
+be replaced.
+
+### Prototype Pollution Attacks (CWE-1321)
+
+Prototype pollution refers to the possibility to modify or inject properties
+into Javascript language items by abusing the usage of _\_proto\__,
+_constructor_, and _prototype_.
+
+Example:
+
+```js
+const a = {"a": 1, "b": 2};
+const data = JSON.parse('{"__proto__": { "polluted": true}}');
+
+const c = Object.assign({}, a, data);
+console.log(c.polluted); // true
```

Comment:
This likely should also mention a potential DOS by doing so people don't think the prototype will always extend `Object.prototype`:

```js
const data2 = JSON.parse('{"__proto__": null}');
const d = Object.assign(a, data2);
d.hasOwnProperty('b'); // Uncaught TypeError: d.hasOwnProperty is not a function
```